rust/phoenix-channel/src/lib.rs

```diff
@@ -33,6 +33,63 @@ pub struct PhoenixChannel<TInboundMsg, TOutboundRes> {
     pending_join_requests: HashSet<OutboundRequestId>,
 }
 
+/// Creates a new [PhoenixChannel] to the given endpoint and waits for an `init` message.
+///
+/// The provided URL must contain a host.
+/// Additionally, you must already provide any query parameters required for authentication.
+#[tracing::instrument(level = "debug", skip(payload, secret_url))]
+#[allow(clippy::type_complexity)]
+pub async fn init<TInitM, TInboundMsg, TOutboundRes>(
```

Comment:
Since we have `init` and we don't use it outside of this module, we should make `connect` private,

Reply:
Hmm, I kind of wanted it to be separate because `init` is our flavour of phoenix-channels and the rest of the module is actually general purpose.